src/main/scala/mesosphere/marathon/api/HttpTransferMetricsHandler.scala

```diff
@@ -115,7 +115,7 @@ object HttpTransferMetricsHandler {
     * these metrics are tracked another way.
     */
   def exclude(request: ServletRequest) = {
-    val state = request match {
+    val state = unwrapRequest(request) match {
       case r: Request => r.getHttpChannelState.getState
       case _ => throw new IllegalStateException("We should never get here")
```

Comment:
Should we change the message? Some more context would be nice. Eg logging `request` or its type.

Reply:
Yes, we should. I did it in another change, but forgot to forward port it.